app.rb

```diff
@@ -0,0 +1,118 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'students'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    return puts 'No books found' if @books.empty?
+
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    return puts 'No people found' if @people.empty?
+
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    if person_type != '1' && person_type != '2'
+      puts 'Invalid option'
+      return
+    end
+
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    if person_type == '1'
+      create_student(age, name)
+    else
+      create_teacher(age, name)
+    end
+
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] You did a good job working on the previous reviewer's suggestions! 💯 However, I'm still getting errors when I try to create a student or a teacher. Please make sure to fix it!

![Screenshot from 2023-06-24 07-17-14](https://github.com/hasnatali1947/-OOP_school_library/assets/77069011/a05e32c7-8373-469e-a729-f9f148c5b5a5)
![Screenshot from 2023-06-24 07-18-21](https://github.com/hasnatali1947/-OOP_school_library/assets/77069011/766238f8-8f3c-41ac-a16a-7526dcd15dff)

💡 _Hint: Read about named parameters, pay special attention to how you're defining your `parent_permission` in your person class as a symbol, and how you're passing it in the `super()` method of each the `teacher` and `student` classes.
Also, in your teacher class, you're sending an extra parameter (id) in the `super()` method. Please check this!_